### DISCUSSION TITLE
Is setting `locale` required if you just want to use the device's locale?

### DISCUSSION THREAD

**Discussion by simon-assistiq** [henninghall/react-native-date-picker#856] [Q&A]
Does the component not automatically pick up the device's locale / language? 

I have set my device's language to French. (The region is set to Canada.) But the date picker shows English text. Even if I change the region to e.g. France, it still shows English months / text (though it does update the order of the date fields). 

```
<DatePicker
  modal
  open={isDatePickerVisible}
  date={value}
  mode="date"
  minimumDate={tomorrow}
  onConfirm={handleConfirm}
  timeZoneOffsetInMinutes={new Date().getTimezoneOffset()}
  onCancel={() => setDatePickerVisible(false)}
/>
```

Is it expected that it doesn't pick up the language automatically?

The main reason why I am asking is because the README doesn't specify any default fallback for the `locale` property. I would have assumed the default would be to use the device's settings.

EDIT:
Even if I pass `locale='fr_CA'`, the months are shown in French but it still says "Select date" at the top and "Confirm" (i.e. English) at the bottom… But I guess I can set the text for `title` and `confirmText` to get around that.

I am using 5.0.7 + React Native 0.73.6 on iOS 17.6.1

## Related
* #828 looks related but there was no response and the poster closed it soon after

**Comment by henninghall** [comment]
Hi, your conclusions in the edit are correct. But, I would like it to pick up French as soon as device language is French and locale prop is unset. Could you send some screenshots of the device settings?

**Reply by simon-assistiq** [reply]
<img src="https://github.com/user-attachments/assets/7ee149b0-136f-4546-ab58-efc41cbe99d9" width="50%">

**Reply by simon-assistiq** [reply]
I just noticed that when I used `toLocaleString(undefined…` in my RN app I also got unexpected results with it not localizing properly. So maybe this date picker library is not the issue here and it's something with React Native not communicating the locale properly?

I did notice that if I try `NativeModules.SettingsManager.settings.AppleLocale` I get `fr_CA` whereas JS seems to be expecting `fr-CA` (plus if I try to pass `fr_CA` to `toLocaleString` it crashes because it doesn't recognize the locale ID). I am not sure if that's at all related.
